kola/harness.go

```diff
@@ -169,15 +170,24 @@ func NewFlight(pltfrm string) (flight platform.Flight, err error) {
 		flight, err = packet.NewFlight(&PacketOptions)
 	case "qemu":
 		flight, err = qemu.NewFlight(&QEMUOptions)
+	case "unprivileged-qemu":
+		flight, err = unprivqemu.NewFlight(&QEMUOptions)
 	default:
 		err = fmt.Errorf("invalid platform %q", pltfrm)
 	}
 	return
 }
 
-func filterTests(tests map[string]*register.Test, pattern, platform string, version semver.Version) (map[string]*register.Test, error) {
+func filterTests(tests map[string]*register.Test, pattern, pltfrm string, version semver.Version) (map[string]*register.Test, error) {
 	r := make(map[string]*register.Test)
 
+	checkPlatforms := []string{pltfrm}
+
+	// unprivileged-qemu has the same restrictions as QEMU but might also want additional restrictions due to the lack of a Local cluster
+	if pltfrm == "unprivileged-qemu" {
```

Comment:
nit for later, not this PR, we should define these as `const` strings in each platform.